AttributedLabel+Link.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+
+extension AttributedLabel {
+    public enum LinkDetectionType: Equatable, Hashable {
+        case date
+        case address
+        case link
+        case phoneNumber
+
+        var checkingType: NSTextCheckingResult.CheckingType {
+            switch self {
+            case .date: return .date
+            case .address: return .address
+            case .link: return .link
+            case .phoneNumber: return .phoneNumber
+            }
+        }
+    }
+
+    public struct Link: Equatable, Hashable {
+        internal var perform: () -> Void
+
+        public private(set) var hashValue: Int
```

Comment:
What's the point of this vs storing and hashing the URL?

Reply:
There's an init that takes in a closure instead of a URL and uses a UUID - I forgot how hashing works though, need to change this